### PR TITLE
Fix issues for Rails 7.2

### PR DESCRIPTION
**Pull request by FLeinzi**

### CODE REVIEW

**Comment by kratob** [comment]
I seem to have fixed it, but the whole approach of the `NoTable` module is becoming unsustainable. :-(

**Comment by kratob** [comment]
@FLeinzi maybe you can take a look. I cannot request a review since you own the PR.

File: lib/active_type/no_table.rb [hunk 1]
```diff
@@ -148,6 +152,12 @@ def _query_by_sql(*)
           []
         end
 
+        def cached_find_by(*)
+        end
```
Comment (FLeinzi) [inline lib/active_type/no_table.rb:156]: Shouldn't we explicitly return `nil` or `[]` (like in the methods above) here?

File: lib/active_type/no_table.rb [hunk 2]
```diff
@@ -132,6 +132,10 @@ def connection
           @connection ||= DummyConnection.new(nil)
         end
 
+        def with_connection(**)
+          yield(connection)
+        end
```
Comment (FLeinzi) [inline lib/active_type/no_table.rb:137]: Nice, this way we prevent to use the connection pool, right?

**Comment by kratob** [comment]
This should fix #190.

**Pull request merged**